pocket_coffea/utils/plot_utils.py

```diff
@@ -1048,44 +1128,145 @@ def _get_err2(self, stacks):
+            hep.cms.lumitext(
+                text=toplabel,
+                fontsize=self.style.fontsize,
+                ax=self.ax,
+            )
+        return self.fig, axes
+
+    def format_figure(self, ratio=True):
+        """Formats the figure's axes, labels, ticks, xlim and ylim."""
+        self.ax.set_ylabel("Counts", fontsize=self.style.fontsize)
+        self.ax.legend(fontsize=self.style.fontsize, ncol=2, loc="upper right")
+        self.ax.tick_params(axis="x", labelsize=self.style.fontsize)
+        self.ax.tick_params(axis="y", labelsize=self.style.fontsize)
+        self.ax.set_xlim(
+            self.style.opts_axes["xedges"][0], self.style.opts_axes["xedges"][-1]
+        )
+        if self.log:
+            self.ax.set_yscale("log")
+            self.ax.set_ylim(
+                (0.01, 10 ** (math.floor(math.log(self.nominal.max(), 10)) + 3))
+            )
+        else:
+            self.ax.set_ylim((0, 1.5*self.nominal.max()))
+
+        if ratio:
+            self.rax.set_xlabel(
+                self.style.opts_axes["xlabel"], fontsize=self.style.fontsize
+            )
+            self.rax.set_ylabel("Shift / Nominal", fontsize=self.style.fontsize)
+            self.rax.yaxis.set_label_coords(-0.075, 1)
+            self.rax.tick_params(axis="x", labelsize=self.style.fontsize)
+            self.rax.tick_params(axis="y", labelsize=self.style.fontsize)
+            self.rax.set_ylim((0.5, 1.5))
+
+        if self.style.has_labels:
+            handles, labels = self.ax.get_legend_handles_labels()
+            labels_new = []
+            handles_new = []
+            for i, label in enumerate(labels):
+                # If additional scale is provided, plot it on the legend:
+                scale_str = ""
+                if self.style.has_rescale_samples and label in self.style.rescale_samples:
+                    scale_str = f" x{self.style.rescale_samples[label]:.2f}"
+                if label in self.style.labels_mc:
+                    
+                    labels_new.append(f"{self.style.labels_mc[label]}"+scale_str)
+                else:
+                    labels_new.append(label+scale_str)
+                    
+                handles_new.append(handles[i])
+            labels = labels_new
+            handles = handles_new
+            self.ax.legend(
+                handles,
+                labels,
+                fontsize=self.style.fontsize,
+                ncol=2,
+                loc="upper right",
+            )
+
+    def plot(self, ratio=True, toplabel=None, log=False):
+        """Plots the nominal, up and down systematic variations on the same plot."""
+
+        # setup figure and corresponding axes
+        self.log = log
+        self.define_figure(ratio=ratio, toplabel=toplabel)
+        
+        # plot histograms
+        # https://stackoverflow.com/a/65935165
         self.ax.hist(
-            self.style.opts_axes["xcenters"],
+            self.bins[:-1],
```

Comment:
before the change the number of bins was set to the default of matplotlib (10 at the time of writing) and the entries in `self.style.opts_axes["xcenters"]` were passed to `numpy.histogram` with 10 bins, which is not what we want.
We want to fill each bin with exactly 1 entry that has a weight according to the histogram (nominal/up/down).
(https://matplotlib.org/3.9.0/api/_as_gen/matplotlib.pyplot.hist.html)